File: foxy/parent/src/command/vanilla/entertainment/CancelExecutor.ts
import { bot } from "../../../FoxyLauncher";
import { ExtendedUser } from "../../../structures/types/DiscordUser";
import { ExecutorParams } from "../../structures/CommandExecutor";

export default class CancelExecutor {
    async execute({ context, endCommand, t }: ExecutorParams) {
        const user = await context.getOption<ExtendedUser>('user', 'users');
        const content = context.interaction ? await context.getOption<string>('reason', false) : context.getMessage(2, true);

        if (!user) {
            context.reply({
                content: context.makeReply(bot.emotes.FOXY_DRINKING_COFFEE, t('commands:cancel.noUser'))
            });
            return endCommand();
        }

        if (!content) {
            context.reply({
                content: context.makeReply(bot.emotes.FOXY_RAGE, t('commands:cancel.noReason'))
            });
            return endCommand();
        }

        context.reply({
            content: context.makeReply(bot.emotes.FOXY_SCARED, t('commands:cancel.result', { user: await bot.rest.foxy.getUserDisplayName((await context.getAuthor()).id), reason: content, mention: user.asMention }))
        })

        return endCommand();
    }
};

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/entertainment/CancelExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.entertainment

import net.cakeyfox.common.FoxyEmotes
import net.cakeyfox.foxy.interactions.commands.CommandContext
import net.cakeyfox.foxy.interactions.commands.UnleashedCommandExecutor
import net.cakeyfox.foxy.interactions.pretty
import net.dv8tion.jda.api.entities.User

class CancelExecutor : UnleashedCommandExecutor() {
    override suspend fun execute(context: CommandContext) {
        context.defer()

        val user = context.getOption("user", 0, User::class.java)
        val reason = context.getOption("reason", 1, String::class.java, true)

        if (user == null || reason == null) return

        context.reply {
            content = pretty(
                FoxyEmotes.FoxyDrinkingCoffee,
                context.locale["cancel.userCancelledBecause", user.asMention, reason]
            )
        }
    }
}